scripts/locale-checks.ts

```diff
@@ -0,0 +1,318 @@
+  return missingItem.length ? missingItem : false;
+};
+
+const findMissingInArrayWithNamespacedLocaleKeys = (
+  computedNamespacedKeys: string[],
+  locale: Locale,
+  option?: { reverse?: boolean; localeKeyNamesOnly?: boolean },
+): StringArrayRecord | false => {
+  const groupByNamespace = computedNamespacedKeys.reduce<StringSetRecord>((acc, item) => {
+    const namespace = getNamespace(item);
+    if (!acc[namespace]) acc[namespace] = new Set();
+    acc[namespace].add(removeNamespace(item));
+    return acc;
+  }, {});
+
+  return Object.entries(locale)
+    .filter(([key]) => key != 'id')
+    .filter(([key]) => !option?.localeKeyNamesOnly || (option?.localeKeyNamesOnly && key === 'names'))
+    .reduce<StringArrayRecord | false>((acc, [namespace, obj]) => {
+      const props = getNestedPropertyNames(obj);
+      const missingItem = option?.reverse
+        ? props.filter((item) => !Array.from(groupByNamespace[namespace] ?? []).includes(item))
+        : Array.from(groupByNamespace[namespace] ?? []).filter((item) => !props.includes(item));
+      if (missingItem.length) {
+        if (acc === false) acc = {};
+        acc[namespace] = missingItem;
+      }
+      return acc;
+    }, false);
+};
+
+const findMissingLocalizedItems = (
+  localeKey: string,
+  localeComputedKeys: string[] | Set<string>,
+  option?: { reverse?: boolean; localeKeyNamesOnly?: boolean },
+): StringArrayRecord | false => {
+  // For a region locale (XxXx), merge it with the base locale (Xx) if exists
+  const locale =
+    localeKey.length === 4 && Object.prototype.hasOwnProperty.call(locales, localeKey.slice(0, 2))
+      ? merge(locales[localeKey.slice(0, 2)], locales[localeKey])
+      : locales[localeKey];
+
+  return findMissingInArrayWithNamespacedLocaleKeys(Array.from(localeComputedKeys), locale, option);
+};
+
+const devLocale = { id: 'dev' };
+const allCalendars: (typeof CalendarDef)[] = [GeneralRoman, ...Object.values(particularCalendars)];
+const allMartyrologyKeys = new Set<string>();
+const allLocalesKeys = new Set<string>();
+const allCalendarData: Record<string, CalendarData> = {};
+
+const metaI18nKeys: string[] = [
+  ...COLORS.map((c) => `colors:${c.toLowerCase()}`),
+  ...SEASONS.map((s) => `seasons:${s.toLowerCase()}.season`),
+  ...PROPER_CYCLES.map((c) => `cycles:${c.toLowerCase()}`),
+  ...SUNDAY_CYCLES.map((c) => `cycles:sunday_${c.toLowerCase()}`),
+  ...WEEKDAY_CYCLES.map((c) => `cycles:weekday_${c.toLowerCase()}`),
+  ...PSALTER_WEEKS.map((c) => `cycles:psalter_${c.toLowerCase()}`),
+  ...RANKS.map((r) => `ranks:${r.toLowerCase()}`),
+  ...MONTHS.map((_m, i) => `months:${i}`),
+  ...WEEKDAYS.map((_w, i) => `weekdays:${i}`),
+];
+
+/**
+ * Compute data to check
+ */
+for (let i = 0; i < allCalendars.length; i++) {
+  const calendar = allCalendars[i];
+
+  // Init config
+  const isGRC = calendar.name === GeneralRoman.name;
+  const builder = isGRC ? new RomcalBuilder(devLocale) : new RomcalBuilder(devLocale, calendar);
```

Comment:
Can’t `calendar` be set to to `general`? :thinking:

Reply:
FYI, we're using an internal Romcal component here. The GRC is already integrated into the calendar computation, so there's no need to reintroduce it at this step.